src/test/java/org/zalando/nakadi/service/EventStreamTest.java

```diff
@@ -327,16 +332,16 @@ private static NakadiKafkaConsumer endlessDummyConsumer() throws NakadiException
     }
 
     private static String jsonBatch(final String partition, final String offset,
-                                    final Optional<List<String>> eventsOrNone) {
+                                    final Optional<List<byte[]>> eventsOrNone) {
         return jsonBatch(partition, offset, eventsOrNone, Optional.empty());
     }
 
     private static String jsonBatch(final String partition, final String offset,
-                                    final Optional<List<String>> eventsOrNone, final Optional<String> metadata) {
+                                    final Optional<List<byte[]>> eventsOrNone, final Optional<String> metadata) {
         final String eventsStr = eventsOrNone
                 .map(events -> {
                     final StringBuilder builder = new StringBuilder(",\"events\":[");
-                    events.forEach(event -> builder.append("\"").append(event).append("\","));
+                    events.forEach(event -> builder.append("\"").append(new String(event)).append("\","));
```

Comment:
I passed UTF-8 Charset to the constructor for now. What kind of test would you like to add?